gencert-private.sh

```diff
@@ -1,15 +1,35 @@
 #!/bin/sh
 
+csr="csr.csr"
+key="key.pem"
+
+# check organizationIdentifier valid value (VATID-XXX... or CF:IT-XXX..)
+
+if echo "$ORGANIZATION" | grep -qe 'CF:IT-*' || echo "$ORGANIZATION" | grep -qe 'VATID-*'; then  
+    echo $ORGANIZATION valid value
+else
+    echo $ORGANIZATION not valid value
+    exit
+fi
+
 # generate configuration file
 
 openssl_conf=$(mktemp)
 
+if [ $(openssl version | grep -c "OpenSSL 1.0") -ge 1 ]; then
+    ORGID_OID="organizationIdentifier=2.5.4.97"
+    ORGID_LABEL="2.5.4.97 organizationIdentifier organizationIdentifier"
+else
+    ORGID_OID=""
+    ORGID_LABEL=""
+fi
+
 cat > "${openssl_conf}" <<EOF
 oid_section=spid_oids
 
 [ req ]
-default_bits=2048
-default_md=sha256
+default_bits=${KEY_LENGTH:=2048}
+default_md=${MD_ALG:=sha256}
```

Comment:
A bit of validation for `KEY_LENGTH` and `MD_ALG` should be used. Take a look at

https://github.com/italia/spid-compliant-certificates/blob/5f4e0e29619223be7a2d47b02fb30c50af37346d/gencert-public.sh#L9-L19